File: backend/src/main/java/com/electoral/entities/Cargo.java
package com.electoral.entities;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "cargos")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Cargo {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String nombre;

    @Column(length = 255)
    private String descripcion;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "elecciones_id", nullable = false)
    private Eleccion elecciones;

    @Enumerated(EnumType.STRING)
    @Column(name = "tipo_votacion", length = 20)
    private TipoVotacion tipoVotacion;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "tipo_circunscripcion_id")
    private TipoCircunscripcion tipoCircunscripcion;

    @Column(name = "cantidad_dignidades")
    @Builder.Default
    private Integer cantidadDignidades = 1;

    @Column(name = "max_candidatos_lista")
    private Integer maxCandidatosLista;

    @Builder.Default
    @Column(nullable = false)
    private Boolean activo = true;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}

File: conteo_electoral_app/lib/models/cargo.dart
class Cargo {
  final int id;
  final String nombre;
  final String descripcion;
  final int eleccionesId;
  final String? tipoVotacion;
  final int? tipoCircunscripcionId;
  final String? tipoCircunscripcionCodigo;
  final String? tipoCircunscripcionNombre;
  final int? cantidadDignidades;
  final int? maxCandidatosLista;
  final bool? activo;

  Cargo({
    required this.id,
    required this.nombre,
    required this.descripcion,
    required this.eleccionesId,
    this.tipoVotacion,
    this.tipoCircunscripcionId,
    this.tipoCircunscripcionCodigo,
    this.tipoCircunscripcionNombre,
    this.cantidadDignidades,
    this.maxCandidatosLista,
    this.activo,
  });

  factory Cargo.fromJson(Map<String, dynamic> json) {
    return Cargo(
      id: json['id'],
      nombre: json['nombre'],
      descripcion: json['descripcion'] ?? '',
      eleccionesId: json['eleccionesId'],
      tipoVotacion: json['tipoVotacion'],
      tipoCircunscripcionId: json['tipoCircunscripcionId'],
      tipoCircunscripcionCodigo: json['tipoCircunscripcionCodigo'],
      tipoCircunscripcionNombre: json['tipoCircunscripcionNombre'],
      cantidadDignidades: json['cantidadDignidades'],
      maxCandidatosLista: json['maxCandidatosLista'],
      activo: json['activo'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nombre': nombre,
      'descripcion': descripcion,
      'eleccionesId': eleccionesId,
      'tipoVotacion': tipoVotacion,
      'tipoCircunscripcionId': tipoCircunscripcionId,
      'tipoCircunscripcionCodigo': tipoCircunscripcionCodigo,
      'tipoCircunscripcionNombre': tipoCircunscripcionNombre,
      'cantidadDignidades': cantidadDignidades,
      'maxCandidatosLista': maxCandidatosLista,
      'activo': activo,
    };
  }
}
